output/elasticsearch/elasticsearch.go

```diff
@@ -0,0 +1,217 @@
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package elasticsearch
+
+//func something() {
+/*if !cfg.HasField("bulk_max_size") {
+	cfg.SetInt("bulk_max_size", -1, defaultBulkSize)
+}*/
+
+/*index, pipeline, err := buildSelectors(im, beat, cfg)
+if err != nil {
+	return outputs.Fail(err)
+}
+
+config := defaultConfig
+if err := cfg.Unpack(&config); err != nil {
+	return outputs.Fail(err)
+}
+
+policy, err := newNonIndexablePolicy(config.NonIndexablePolicy)
+if err != nil {
+	log.Errorf("error while creating file identifier: %v", err)
+	return outputs.Fail(err)
+}
+
+hosts, err := outputs.ReadHostList(cfg)
+if err != nil {
+	return outputs.Fail(err)
+}
+
+if proxyURL := config.Transport.Proxy.URL; proxyURL != nil && !config.Transport.Proxy.Disable {
+	log.Debugf("breaking down proxy URL. Scheme: '%s', host[:port]: '%s', path: '%s'", proxyURL.Scheme, proxyURL.Host, proxyURL.Path)
+	log.Infof("Using proxy URL: %s", proxyURL)
+}
+
+params := config.Params
+if len(params) == 0 {
+	params = nil
+}
+
+if policy.action() == dead_letter_index {
+	index = DeadLetterSelector{
+		Selector:        index,
+		DeadLetterIndex: policy.index(),
+	}
+}
+
+clients := make([]outputs.NetworkClient, len(hosts))
+for i, host := range hosts {
+	esURL, err := common.MakeURL(config.Protocol, config.Path, host, 9200)
+	if err != nil {
+		log.Errorf("Invalid host param set: %s, Error: %+v", host, err)
+		return outputs.Fail(err)
+	}
+
+	var client outputs.NetworkClient
+	client, err = NewClient(ClientSettings{
+		ConnectionSettings: eslegclient.ConnectionSettings{
+			URL:              esURL,
+			Beatname:         beat.Beat,
+			Kerberos:         config.Kerberos,
+			Username:         config.Username,
+			Password:         config.Password,
+			APIKey:           config.APIKey,
+			Parameters:       params,
+			Headers:          config.Headers,
+			CompressionLevel: config.CompressionLevel,
+			Observer:         observer,
+			EscapeHTML:       config.EscapeHTML,
+			Transport:        config.Transport,
+		},
+		Index:              index,
+		Pipeline:           pipeline,
+		Observer:           observer,
+		NonIndexableAction: policy.action(),
+	}, &connectCallbackRegistry)
+	if err != nil {
+		return outputs.Fail(err)
+	}
+
+	client = outputs.WithBackoff(client, config.Backoff.Init, config.Backoff.Max)
+	clients[i] = client
+}
+
+return outputs.SuccessNet(config.LoadBalance, config.BulkMaxSize, config.MaxRetries, clients)
+*/
+//}
+
+/*func init() {
+	outputs.RegisterType("elasticsearch", makeES)
+}*/
+
+const logSelector = "elasticsearch"
+
+/*func makeES(
+	im outputs.IndexManager,
+	beat beat.Info,
+	observer outputs.Observer,
+	cfg *config.C,
+) (outputs.Group, error) {
+	log := logp.NewLogger(logSelector)
+	if !cfg.HasField("bulk_max_size") {
+		cfg.SetInt("bulk_max_size", -1, defaultBulkSize)
+	}
+
+	index, pipeline, err := buildSelectors(im, beat, cfg)
```

Comment:
I think this is for ingest pipelines, and in the case of Agent these are configured by Fleet using index templates and pattern matching against the data stream name: https://www.elastic.co/guide/en/elasticsearch/reference/8.4/ingest.html#pipelines-for-fleet-elastic-agent

I don't think we need to keep this in the shipper.